src/safeposix/mod.rs

```diff
@@ -1,3 +1,28 @@
+//! This module handles system call requests from the Native Client in the RustPOSIX environment.
+//! 
+//! ## top-level features:
+//!
+//! - ### Dispatcher/RPC: 
+//!     - The dispatcher receives system call requests from Native Client. It checks if the cage exists in the cage table, and if it doesn't, initializes a new cage. It then takes the cage object corresponding to that ID number, and calls the method corresponding to the sent call number.
+//! 
+//! - ### Cage Objects:
+//! 
+//!     - Each cage object has a Cage ID, Current Working Directory, Parent ID, and a File Descriptor Table.
+//!
+//! - ### File Descriptor Table: 
+//!     - The file descriptor table is a hash map of file descriptor integers to our file descriptor representations. File Descriptors are implemented as an Enum that can correspond to five descriptor types (File, Stream, Socket, Pipe, Epoll).
+//!
+//! - ### System Calls: 
+//!     - Each cage object has public methods corresponding to each system call. These calls are implemented either as filesystem related calls, system related calls, or network related calls in their respective files.
+//!
+//! - ### FS Metadata: 
+//!     - Our metadata table is similar to how the metadatatable from SafePOSIX-Repy operated. The table is represented by a struct with fields: nextinode, dev_ud, inodetable. The Inode Enum can describe a variety of Inode structs which include: File(generic), CharDev, Socket, Directory.
```

Comment:
I'm not sure you need this first sentence. Most people won't know what this means.

Reply:
will refactor